LoopStructural/modelling/intrusions/intrusion_body.py

```diff
@@ -0,0 +1,1118 @@
+        radius : The maximum isotropic search radius.
+
+        Returns
+        s_parameters : dictionary of parameters to be used for lateral extent simulation. Parameters definition above
+        ----
+        """
+        if "tmin" in lateral_simulation_parameters:
+            tmin = lateral_simulation_parameters["tmin"]
+        else:
+            tmin = -999
+
+        if "tmax" in lateral_simulation_parameters:
+            tmax = lateral_simulation_parameters["tmax"]
+        else:
+            tmax = 999
+
+        if "itrans" in lateral_simulation_parameters:
+            itrans = lateral_simulation_parameters["itrans"]
+        else:
+            itrans = 1
+
+        if "ktype" in lateral_simulation_parameters:
+            ktype = lateral_simulation_parameters["ktype"]
+        else:
+            ktype = 0
+
+        if "nx" in lateral_simulation_parameters:
+            nx = lateral_simulation_parameters["nx"]
+        else:
+            nx = 200
+
+        if "ny" in lateral_simulation_parameters:
+            ny = lateral_simulation_parameters["ny"]
+        else:
+            ny = 3
+
+        if "xmn" in lateral_simulation_parameters:
+            xmn = lateral_simulation_parameters["xmn"]
+        else:
+            xmn = None
+
+        if "ymn" in lateral_simulation_parameters:
+            ymn = lateral_simulation_parameters["ymn"]
+        else:
+            ymn = -0.0001
+
+        if "xsiz" in lateral_simulation_parameters:
+            xsiz = lateral_simulation_parameters["xsiz"]
+        else:
+            xsiz = None
+
+        if "ysiz" in lateral_simulation_parameters:
+            ysiz = lateral_simulation_parameters["ysiz"]
+        else:
+            ysiz = 0.0001
+
+        if "zmin" in lateral_simulation_parameters:
+            zmin = lateral_simulation_parameters["zmin"]
+        else:
+            zmin = None
+
+        if "zmax" in lateral_simulation_parameters:
+            zmax = lateral_simulation_parameters["zmax"]
+        else:
+            zmax = None
+
+        if "nxdis" in lateral_simulation_parameters:
+            nxdis = lateral_simulation_parameters["nxdis"]
+        else:
+            nxdis = 1
+
+        if "nydis" in lateral_simulation_parameters:
+            nydis = lateral_simulation_parameters["nydis"]
+        else:
+            nydis = 1
+
+        if "ndmin" in lateral_simulation_parameters:
+            ndmin = lateral_simulation_parameters["ndmin"]
+        else:
+            ndmin = 0
+
+        if "ndmax" in lateral_simulation_parameters:
+            ndmax = lateral_simulation_parameters["ndmax"]
+        else:
+            ndmax = 3
+
+        if "radius" in lateral_simulation_parameters:
+            radius = lateral_simulation_parameters["radius"]
+        else:
+            radius = 500
+
+        s_parameters = {
```

Comment:
You can probably simplify all of this by replacing the tmin,tmax variables with the kwarg.get('key',default_val) for constructing this dictionary

Reply:
done